### ISSUE TITLE
Missing cells, problem with colspan?

### ISSUE THREAD

**Issue by OliverColeman** [Ticketpark/HtmlPhpExcel#7]
I tried converting the table below and found many cells were not included in the output. I'm guessing there are some problems with handling of colspan? I've attached the output xlsx.

The PHP code:
```
  $htmlPhpExcel = new \Ticketpark\HtmlPhpExcel\HtmlPhpExcel($table);
  $objPHPExcel = $htmlPhpExcel->process()->getExcelObject();
  
  $excel_title = preg_replace('/[\[\]:?!/\\*]/', '', $title);
  wpr("excel title: $excel_title");
  $objPHPExcel->getProperties()->setTitle($excel_title)->setCreator("Naplan Online Wizard");
  $objPHPExcel->getActiveSheet()->setTitle(t("Class results"));
  
  $writer = PHPExcel_IOFactory::createWriter($objPHPExcel, 'Excel2007');
  $writer->save($tmpfile);
```

The table:
```
[ojc_test_quiz-1035-20170321_074330.xlsx](https://github.com/Ticketpark/HtmlPhpExcel/files/856339/ojc_test_quiz-1035-20170321_074330.xlsx)

<table id="class-results-table">
<tbody>
 <tr class="odd"><th colspan="2">Student</th><th colspan="2">Result</th><th colspan="2" class="area">Long Vowels</th><th colspan="1" class="area">Prepositions and Conjunctions</th><th colspan="1" class="area">Silent e</th> </tr>
 <tr class="even"><th>Last</th><th>First</th><th>Correct</th><th>Score</th><th><span class="question-number init-coapopup-processed" data-nid="836446">2</span></th><th><span class="question-number init-coapopup-processed" data-nid="836451">3</span></th><th><span class="question-number init-coapopup-processed" data-nid="836236">4</span></th><th><span class="question-number init-coapopup-processed" data-nid="842183">1</span></th> </tr>
 <tr class="odd"><td class="student-name"><a href="/codev2/assessment/836470/results/140?iframe=true&amp;width=1000&amp;height=10000" class="colorbox-load init-colorbox-load-processed cboxElement">1</a></td><td class="student-name"><a href="/codev2/assessment/836470/results/140?iframe=true&amp;width=1000&amp;height=10000" class="colorbox-load init-colorbox-load-processed cboxElement">ojc</a></td><td>1</td><td>25%</td><td><span title="Score: 0%
Student: 1 ojc
[ojc_test_quiz-1035-20170321_074330.xlsx](https://github.com/Ticketpark/HtmlPhpExcel/files/856331/ojc_test_quiz-1035-20170321_074330.xlsx)

Area: Long Vowels
Question: 2
">✗</span></td><td><span title="Score: 0%
Student: 1 ojc
Area: Long Vowels
Question: 3
">✗</span></td><td><span title="Score: 0%
Student: 1 ojc
Area: Prepositions and Conjunctions
Question: 4
">✗</span></td><td><span title="Score: 100%
Student: 1 ojc
Area: Silent e
Question: 1
">✓</span></td> </tr>
 <tr class="even"><td class="student-name">ggf</td><td class="student-name">fgrd</td><td>0</td><td>0%</td><td><span title="Score: 0%
Student: ggf fgrd
Area: Long Vowels
Question: 2
">✗</span></td><td><span title="Score: 0%
Student: ggf fgrd
Area: Long Vowels
Question: 3
">✗</span></td><td><span title="Score: 0%
Student: ggf fgrd
Area: Prepositions and Conjunctions
Question: 4
">✗</span></td><td><span title="Score: 0%
Student: ggf fgrd
Area: Silent e
Question: 1
">✗</span></td> </tr>
 <tr class="odd"><th colspan="4" class="column_aggregate_header">Class correct total: </th><td class="column_aggregate">0</td><td class="column_aggregate">0</td><td class="column_aggregate">0</td><td class="column_aggregate">1</td> </tr>
 <tr class="even"><th colspan="4" class="column_aggregate_header">Class correct percent: </th><td class="column_aggregate">0%</td><td class="column_aggregate">0%</td><td class="column_aggregate">0%</td><td class="column_aggregate">50%</td> </tr>
 <tr class="odd"><th colspan="4" class="column_aggregate_header">Class area correct percent: </th><td colspan="2" class="area-score column_aggregate">0%</td><td colspan="1" class="area-score column_aggregate">0%</td><td colspan="1" class="area-score column_aggregate">50%</td> </tr>
</tbody>
</table>
```

**Comment by OliverColeman** [comment]
Okay, here's a link to the xlsx file, the upload didn't work.. https://drive.google.com/open?id=0B_81N_5rBw__SzVBUXlRcWpRQTg
Oh yeah, the character encoding for the ticks and crosses isn't working properly, but that's my problem that I'm yet to fix, not a problem with this library :).

**Comment by OliverColeman** [comment]
Found the problem. In HtmlPhpExcel::createExcel(), the $cellNumber variable needs to be incremented according to the colspan, eg around line 286 replace:
```
                        if ($colspan) {$colspan = $colspan - 1;}
                        if ($rowspan) {$rowspan = $rowspan - 1;}
                        $mergeCellsTargetCellIndex = \PHPExcel_Cell::stringFromColumnIndex($cellNumber + $colspan).($rowNumber + $rowspan);
```
with
```
                        if ($colspan) {
                          $colspan = $colspan - 1;
                          $cellNumber += $colspan;
                        }
                        if ($rowspan) {$rowspan = $rowspan - 1;}
                        $mergeCellsTargetCellIndex = \PHPExcel_Cell::stringFromColumnIndex($cellNumber).($rowNumber + $rowspan);
```
As far as I can tell there will (still) be an issue with rowspans.

**Comment by sprain** [comment]
Maybe connected to #12?

**Comment by sprain** [comment]
Yeah, this looks to be fixed by #12.
